pkg/transport/listen_bsd.go

```diff
@@ -0,0 +1,58 @@
+//go:build freebsd || netbsd || openbsd || dragonfly
+// +build freebsd netbsd openbsd dragonfly
```

Comment:
darwin.go can be merged here too?

Reply:
Seems like it would be easy enough to duplicate this file when BSD-specific support would be needed? I'd prefer to only have `//go:build darwin || dragonfly || freebsd || netbsd || openbsd` for now, instead of adding more duplicate `listen` code.
If you duplicate the code, I'd remove the `case "vsock"` block which is hyperkit specific, and `ListenUnixgram` is also not expected to be used as it's vfkit-specific, so this could error out.

Reply:
Seems like a good reason to have two, one for darwin and one for bsd

Reply:
If the BSD files are different from the darwin files, yes, this is a good reason for having 2 files. As the PR is right now, the files are the same.